### PR TITLE
Tweak std.internal.scopebuffer docs

### PR DESCRIPTION
**Pull request by ntrel**
* Respect `realloc` parameter.
* Fix ddoc for commented out `this(this)` concatenated into `put`'s docs.
* Fix missing ddoc for `put(CT[])` due to private `CT`.
* Use documented unittest for scopeBuffer instead of inline code (which had a typo).

### CODE REVIEW

File: std/internal/scopebuffer.d [hunk 1]
```diff
@@ -152,11 +153,12 @@ struct ScopeBuffer(T, alias realloc = /*core.stdc.stdlib*/.realloc)
     /****************************
      * Copying of ScopeBuffer is not allowed.
      */
-    //@disable this(this);
```
Comment (JackStouffer) [inline std/internal/scopebuffer.d]: Why not just delete the docs and the commented out code here

File: std/internal/scopebuffer.d [hunk 2]
```diff
@@ -178,7 +180,7 @@ struct ScopeBuffer(T, alias realloc = /*core.stdc.stdlib*/.realloc)
      * If $(D const(T)) can be converted to $(D T), then put will accept
      * $(D const(T)[]) as input. It will accept a $(D T[]) otherwise.
      */
-    private alias CT = Select!(is(const(T) : T), const(T), T);
+    package alias CT = Select!(is(const(T) : T), const(T), T);
```
Comment (JackStouffer) [inline std/internal/scopebuffer.d]: rationale?

Comment (ntrel) [inline std/internal/scopebuffer.d]: CT has the docs for the following `put`, which uses `ditto`. As CT is private, ddoc ignores its docs and the ditto makes `put(CT[])` ditto the first overload of `put`. (Maybe ddoc could be changed to be more helpful here).

File: std/internal/scopebuffer.d (hunk 1 above)
Comment (ntrel) [inline std/internal/scopebuffer.d]: now done

**Comment by JackStouffer** [comment]
Auto-merge toggled on

**Pull request merged**